src/app/inventory/store/sockets.ts

```diff
@@ -479,10 +484,51 @@ function buildSocket(
     socketIndex: index,
     plugged,
     plugOptions,
+    plugSet: socketDef.reusablePlugSetHash
+      ? buildCachedDimPlugSet(defs, socketDef.reusablePlugSetHash)
+      : undefined,
     curatedRoll,
     hasRandomizedPlugItems,
     reusablePlugItems: reusablePlugs,
     isPerk,
     socketDefinition: socketDef,
   };
 }
+
+// This cache is used to reuse DimPlugSets across items. If we didn't do this each
+// item would have their own instances of shaders, which is 100's of plugs.
+let reusablePlugSetCache: { [plugSetHash: number]: DimPlugSet | undefined } = {};
+
+/**
+ * Resets the cache we use to ensure plug sets are only created once per hash.
+ */
+export function resetReusablePlugSetCache() {
```

Comment:
If we're really depending on only defs we don't need to reset.

Reply:
Yeah you are right, I built that initially with the intent to build them off live stats. Now that isn't happening I can remove this.